netket/models/autoreg.py

```diff
@@ -278,6 +340,32 @@ def _normalize(log_psi: Array, machine_pow: int) -> Array:
     )
 
 
+@dispatch
+def _conditionals_log_psi(model: ARNNDenseModPhase, inputs: Array) -> Array:
+    """
+    Computes the log of the conditional wave-functions for each site if it takes each value
+    for an ARNN model whose modulus and phase are given by different neural networks.
+    """
+
+    x = jnp.expand_dims(inputs, axis=-1)
+
+    x_imag = jnp.copy(x)
```

Comment:
Do we really need a copy here?

Reply:
I think we do, otherwise we would mix imaginary configurations with real configurations

Reply:
I mean you can just write `x_imag = x`. But anyway, the call to `jnp.copy` will be optimized away by JIT in this case, so it's not a big problem

I noticed it just because in plain numpy if you call `np.copy` it will allocate new memory, which is unnecessary in this case because you don't mutate `x`.